service/s3/body_hash.go

```diff
@@ -90,7 +67,7 @@ func computeBodyHashes(r *request.Request) {
 		dst = io.MultiWriter(hashers...)
 	}
 
-	if _, err := copySeekableBody(dst, r.Body); err != nil {
+	if _, err := util.CopySeekableBody(dst, r.Body); err != nil {
```

Comment:
Not sure that a copy body should be apart of a generic package util bundled with the checksum behavior. Maybe this would be better in the aws package with the other body/seeker utilities.